src/widget/wlibrarytableview.cpp

```diff
@@ -122,19 +122,23 @@ void WLibraryTableView::saveTrackModelState(
     VERIFY_OR_DEBUG_ASSERT(!key.isEmpty()) {
         return;
     }
-    ModelState* state = m_modelStateCache[key];
+    ModelState* state = m_modelStateCache.take(key);
     if (!state) {
         state = new ModelState();
     }
     state->verticalScrollPosition = verticalScrollBar()->value();
     state->horizontalScrollPosition = horizontalScrollBar()->value();
-    const QModelIndexList selectedIndexes = selectionModel()->selectedIndexes();
-    if (!selectedIndexes.isEmpty()) {
-        state->selectionIndex = selectedIndexes;
+    if (!selectionModel()->selectedIndexes().isEmpty()) {
+        state->selectionIndex = selectionModel()->selectedIndexes();
     } else {
         state->selectionIndex = QModelIndexList();
     }
```

Comment:
either use `hasSelection`
but actually `selectionModel()->selectedIndexes()` always returns a QModelIndexList, with zero or more items, so the check is unnecessary